### DISCUSSION TITLE
SSO with OAuth (SAML/OpenID Connect)

### DISCUSSION THREAD

**Discussion by mprajescu** [mediacms-io/mediacms#27] [Feature Requests]
I am continuing the conversation from: https://github.com/mediacms-io/mediacms/issues/13 

Can you please implement OAuth 2 for user registration, user provisioning and SSO etc.
I would really appreciate if I could be able to add my existing users in Keycloak and allow them to use this wonderful application.

https://docs.authlib.org/en/latest/django/index.html

Or even:

https://django-oauth-toolkit.readthedocs.io/en/latest/

mgogoulos commented 1 hour ago

Hi, please move this discussion to ideas section on https://github.com/mediacms-io/mediacms/discussions so we can continue the discussion there.
django-allauth is integrated to handle user registration (providing lots of interesting options with little work involved) so we should see how this could potentially co-exist with these OAuth2 libs...

**Comment by mprajescu** [comment]
I’ve just looked over this project and I find it very interesting to add, better than the toolkit one, it’s called Allauth. 

Link to project: http://github.com/pennersr/django-allauth 

Link to docs: https://django-allauth.readthedocs.io/en/latest/

I’ve implemented this in one of my Django projects and it seems to be fairly easy to add if it’s a new project. 

I don’t know the underlying structure of this app but when I find some time, I’ll take a look and see if can be added easily.

**Comment by rhclayto** [comment]
This would be most desirable.

django-allauth, used by this project, has an OpenID provider feature. However, the documentation leaves much to be desired. I have tried hacking this code to create an OpenID login, but it's just too confusing for me how to get it working. 

Is there some way a discussion can be started about using the OpenID feature, which is already a part of django-allauth, in this project to allow adding OpenID based SSO to this software?

**Comment by mgogoulos** [comment]
SAML is supported through https://github.com/mediacms-io/mediacms/releases/tag/v5.0.0 now :)
